Bluetooth: hci_raw: Fix pushing H4 headers in bt_buf_get_rx

Vudentz opened the pull request:
User of bt_buf_get_rx may attempt to push their own headers, e.g. ACL
headers, so move the H4 header logic to bt_recv after logging into the
monitor since otherwise the buffer would contain H4 headers which is
not expected by the monitor.
    
Fixes #24646

jhedberg commented on samples/bluetooth/hci_uart/src/main.c:
```diff
@@ -110,6 +112,30 @@ static void h4_acl_recv(struct net_buf *buf, int *remaining)
+	bytes = uart_fifo_fill(hci_uart_dev, buf->data, buf->len);
+	net_buf_pull(buf, bytes);
+
+	if (buf->len) {
+		return;
+	}
+
+	net_buf_unref(buf);
+	buf = NULL;
```
I'd just make this:
```
if (!buf->len) {
        net_buf_unref(buf);
        buf = NULL;
}

jhedberg reviewed (changes requested):
One minor stylistic comment, otherwise looks good. Hopefully it's tested?

Vudentz committed:
Bluetooth: hci_raw: Fix pushing H4 headers in bt_buf_get_rx

User of bt_buf_get_rx may attempt to push their own headers, e.g. ACL
headers, so move the H4 header logic to bt_recv after logging into the
monitor since otherwise the buffer would contain H4 headers which is
not expected by the monitor.

Fixes #24646

Signed-off-by: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>

carlescufi reviewed (changes requested):
So, of the 2 commits:

`Bluetooth: hci_raw: Fix pushing H4 headers in bt_buf_get_rx` fixes the issue

`Bluetooth: hci_uart: Use interrupt to trigger TX data` makes `hci_uart` not even respond to the first command.

Seeing how we are close to release, I would suggest dropping `Bluetooth: hci_uart: Use interrupt to trigger TX data`  from this PR. 

@jhedberg @Vudentz @joerchan thoughts?

Vudentz commented:
> So, of the 2 commits:
> 
> `Bluetooth: hci_raw: Fix pushing H4 headers in bt_buf_get_rx` fixes the issue
> 
> `Bluetooth: hci_uart: Use interrupt to trigger TX data` makes `hci_uart` not even respond to the first command.
> 
> Seeing how we are close to release, I would suggest dropping `Bluetooth: hci_uart: Use interrupt to trigger TX data` from this PR.
> 
> @jhedberg @Vudentz @joerchan thoughts?

Sure, the first one was just a tentive to fix the later.

carlescufi reviewed (approved):
Re-tested, works fine, thank you!
